src/lando/utils/management/commands/maintenance.py

```diff
@@ -20,56 +20,59 @@ def add_arguments(self, parser):  # noqa: ANN001
             help="Enter one of on, off",
         )
 
-        parser.add_argument(
-            "-c",
-            "--check-migrations",
-            action="store_true",
-            default=False,
-            help="Do not turn on maintenance if there are no unapplied migrations",
-        )
+    def _pause_workers(self):
+        """Pause all workers."""
+        workers = Worker.objects.all()
+        for worker in workers:
+            worker.pause()
+            self.stdout.write(f"Paused {worker}.")
+
+    def _resume_workers(self):
+        """Resume all workers."""
+        workers = Worker.objects.all()
+        for worker in workers:
+            worker.resume()
+            self.stdout.write(f"Resumed {worker}.")
 
     def _turn_on_maintenance_mode(self):
+        """Put the web interface into maintenance mode."""
         ConfigurationVariable.set(
             ConfigurationKey.API_IN_MAINTENANCE, VariableTypeChoices.BOOL, "1"
         )
-        workers = Worker.objects.all()
-        for worker in workers:
-            worker.is_paused = True
-            worker.save()
+        self.stdout.write("Turned maintenance mode on for web interface.")
 
     def _turn_off_maintenance_mode(self):
+        """Put the web interface back online."""
         ConfigurationVariable.set(
             ConfigurationKey.API_IN_MAINTENANCE, VariableTypeChoices.BOOL, "0"
         )
-        workers = Worker.objects.all()
-        for worker in workers:
-            worker.is_paused = False
-            worker.save()
+        self.stdout.write("Turned maintenance mode off for web interface.")
 
     def handle(self, *args, **options):
-        action_mapping = {
-            "on": self._turn_on_maintenance_mode,
-            "off": self._turn_off_maintenance_mode,
-        }
+        ON = "on"
+        OFF = "off"
+
+        actions = (ON, OFF)
         action = options["action"]
-        check = options["check_migrations"]
-        if action not in action_mapping:
+        if action not in actions:
             raise CommandError(
-                f"Action must be one of: {', '.join(action_mapping.keys())}. "
+                f"Action must be one of: {', '.join(actions)}. "
                 f'"{action}" was provided.'
             )
 
-        if not check:
-            action_mapping[action]()
-            self.stdout.write(f"Turned maintenance mode {action}.")
-            sys.exit()
-
+        # If there are unapplied migrations, then both the web interface and the
+        # workers need to be put in maintenance.
         unapplied_migrations = (
             subprocess.run(["lando", "migrate", "--check", "--noinput"]).returncode != 0
```

Comment:
Unrelated to this PR, but should we use something like `call_command` rather than forking a subprocess? https://docs.djangoproject.com/en/5.0/ref/django-admin/#running-management-commands-from-your-code

Reply:
If I remember correctly, I think I ran into trouble capturing `returncode` from `call_command`, which may be why I reverted to this. I filed a bug to tackle this in the future, I'm sure there's a way to do it using `call_command`.